### PR TITLE
Database and fame chart

### PR DESCRIPTION
**Pull request by BaltaJmn**

### CODE REVIEW

File: core/data/src/main/kotlin/com/wachon/spotiwrap/di/DataModule.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.wachon.spotiwrap.data.UserRepository
+import org.koin.core.module.Module
+import org.koin.dsl.module
+
+val DataModule = module {
+    includes(
+        DatabaseModule,
+        NetworkModule,
+        DataDataModule,
+        DataDomainModule,
```
Comment (Naxomiun) [inline core/data/src/main/kotlin/com/wachon/spotiwrap/di/DataModule.kt]: DataDomain & DataPresentation makes no sense ✓:

File: core/database/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+plugins {
+    spotiwrap_android_library
+    kotlin("kapt")
```
Comment (Naxomiun) [inline core/database/build.gradle.kts:3]: Maybe we could check for KSP impl?

File: core/database/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+dependencies {
+    implementation(project(":core:auth"))
+    implementation(libs.room.runtime)
+    implementation(libs.room.ktx)
+    kapt(libs.room.compiler)
+    implementation(libs.room.rxjava3)
+    implementation(libs.room.guava)
+    implementation(libs.room.paging)
```
Comment (Naxomiun) [inline core/database/build.gradle.kts]: What are RXJava, Guava and Paging for?

File: features/artists/src/main/kotlin/com/wachon/spotiwrap/features/artists/presentation/common/ArtistItem.kt [hunk 4]
```diff
@@ -38,12 +49,52 @@ fun ArtistItem(
...
+                if (artist.artistFame != ItemFame.NONE) {
+                    Row(
+                        horizontalArrangement = Arrangement.End,
+                        verticalAlignment = Alignment.CenterVertically,
+                        modifier = Modifier
+                            .fillMaxSize()
+                    ) {
+                        Image(
+                            when (artist.artistFame) {
+                                ItemFame.UP -> painterResource(R.drawable.fame_up)
```
Comment (Naxomiun) [inline features/artists/src/main/kotlin/com/wachon/spotiwrap/features/artists/presentation/common/ArtistItem.kt]: Maybe can we add the icon resource into the enum as a property?

File: features/tracks/src/main/kotlin/com/wachon/spotiwrap/features/tracks/presentation/common/TrackItem.kt [hunk 5]
```diff
@@ -37,12 +54,51 @@ fun TrackItem(
...
+                modifier = Modifier
+                    .height(25.dp)
+                    .fillMaxWidth()
+                    .background(
+                        Brush.horizontalGradient(
+                            colors = listOf(Color.Transparent, Color.Black), startX = 175f, endX = 240f
+                        )
+                    )
+            ) {
+                if (track.trackFame != NONE) {
```
Comment (Naxomiun) [inline features/tracks/src/main/kotlin/com/wachon/spotiwrap/features/tracks/presentation/common/TrackItem.kt]: This code seems to be repeated twice and maybe its going to be used more. Maybe we could think in making an standalone composable

File: settings.gradle.kts [hunk 6]
```diff
@@ -28,3 +30,4 @@ include(
 include(
     ":test:shared-test"
 )
+include()
```
Comment (Naxomiun) [inline settings.gradle.kts]: Bruh

Comment (BaltaJmn) [reply settings.gradle.kts]: image

File: features/tracks/src/main/kotlin/com/wachon/spotiwrap/features/tracks/presentation/common/TrackItem.kt (hunk 5 above)
Comment (BaltaJmn) [reply features/tracks/src/main/kotlin/com/wachon/spotiwrap/features/tracks/presentation/common/TrackItem.kt]: Component created ;)

File: core/data/src/main/kotlin/com/wachon/spotiwrap/di/DataModule.kt (hunk 1 above)
Comment (BaltaJmn) [reply core/data/src/main/kotlin/com/wachon/spotiwrap/di/DataModule.kt]: Totally agree

File: core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/ArtistsRepository.kt [hunk 7]
```diff
@@ -35,16 +35,37 @@ class DefaultArtistsRepository(
                 timeRange = TopItemTimeRange.MEDIUM_TERM.name.lowercase()
             )
 
-            artistDao.insertArtists(
-                compareFame(apiItems.items ?: emptyList(), artistDao.getArtistsNoFlow())
-                    .mapIndexed { index, topItemApi -> topItemApi.toArtistDB(index) }
-            )
+            val dbItems = artistDao.getArtistsNoFlow()
+
+            if (!isContentTheSame(apiItems.items ?: emptyList(), dbItems)) {
```
Comment (Naxomiun) [inline core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/ArtistsRepository.kt]: I don't understand this 🌵

Comment (BaltaJmn) [reply core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/ArtistsRepository.kt]: Very easy, this method is responsible for checking if what we get from the api and what we have in the database is the same content, to save putting the same data and not always get the equals symbol in the fame item

Comment (Naxomiun) [reply core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/ArtistsRepository.kt]: But... whats the problem? if lists are the same then index are the same and then nothing changes (?)

Comment (BaltaJmn) [reply core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/ArtistsRepository.kt]: Why should it change if it is the same content in the same order?

**Pull request merged**